drivers/docker/fingerprint.go

```diff
@@ -16,6 +17,20 @@ func (d *Driver) Fingerprint(ctx context.Context) (<-chan *drivers.Fingerprint,
 	return ch, nil
 }
 
+// setFingerprintSuccess marks the driver as having fingerprinted successfully
+func (d *Driver) setFingerprintSuccess() {
+	d.fingerprintLock.Lock()
+	d.fingerprintSuccess = helper.BoolToPtr(true)
+	d.fingerprintLock.Unlock()
+}
+
+// setFingerprintFailure marks the driver as having failed fingerprinting
+func (d *Driver) setFingerprintFailure() {
```

Comment:
There is some code duplication here. Maybe have a `success` parameter here and set it to `fingerprintSuccess`?